Test that the swiftinterface for the stdlib can be compiled

jrose-apple opened the pull request:
...like we've done with the overlays for months. Meant to do this a long time ago, but at the time we hadn't gotten the stdlib working yet!

jrose-apple commented:
@swift-ci Please test

harlanhaskins commented on validation-test/ParseableInterface/verify_stdlib.swift:
```diff
@@ -0,0 +1,7 @@
+// been generated.
+
+// RUN: %empty-directory(%t)
+// RUN: test ! -e %platform-module-dir/Swift.swiftmodule/%target-cpu.swiftinterface || %target-swift-frontend -build-module-from-parseable-interface %platform-module-dir/Swift.swiftmodule/%target-cpu.swiftinterface -parse-stdlib -o %t/Swift.swiftmodule
```
@compnerd Is this good on Windows?

jrose-apple replied:
Oops, right. I copied from `verify_simd` but that's guarded by `objc_interop` anyway.

harlanhaskins reviewed (approved):
🎉

jrose-apple commented:
@swift-ci Please test Linux

jrose-apple committed:
Test that the swiftinterface for the stdlib can be compiled

...like we've done with the overlays for months. Meant to do this a
long time ago, but at the time we hadn't gotten the stdlib working
yet!

jrose-apple commented:
@swift-ci Please smoke test
